src/main/kotlin/com/performance/domain/reservation/service/ReservationService.kt

```diff
@@ -13,19 +12,54 @@ import org.springframework.transaction.annotation.Transactional
 class ReservationService(
     private val reservationRepository: ReservationRepository,
     private val performanceRepository: PerformanceRepository,
-    private val seatRepository: SeatRepository
 ) {
     @Transactional
-    fun saveReservation(request: ReservationRequest) =
-        reservationRepository.save(
+    fun saveReservation(request: ReservationRequest): Long {
+        check(!unavailableReservation(request)) {
+            throw IllegalStateException()
+        }
+
+        return reservationRepository.save(
             Reservation(
                 id = request.id,
                 email = request.email,
-                performance = performanceRepository.findByIdOrNull(request.performanceId)
-                    ?: throw IllegalStateException(),
-                seat = seatRepository.findByIdOrNull(request.seatId)
-                    ?: throw IllegalStateException(),
+                performanceId = request.performanceId,
+                seatId = request.seatId,
                 status = request.status
             )
-        )
+        ).id
+    }
+
+    private fun unavailableReservation(request: ReservationRequest): Boolean {
+        val performance =
+            performanceRepository.findByIdOrNull(request.performanceId) ?: throw IllegalStateException()
+        return performance.canReserve() &&
+                reservationRepository.findByPerformanceIdAndSeatId(
+                    request.performanceId,
+                    request.seatId
+                ).isEmpty()
+    }
+
+    @Transactional
+    fun updateReservation(request: ReservationRequest) {
+        val reservation = reservationRepository.findByIdOrNull(request.id)
+            ?: throw IllegalStateException()
```

Comment:
```suggestion
            ?: throw IllegalStateException()
```

CustomException으로 만들어주세요 !